x-pack/plugins/endpoint/public/applications/endpoint/store/routing/middleware.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AppAction } from '../action';
+import { MiddlewareFactory } from '../../types';
+import { flattenedPromise } from '../../lib/flattened_promise';
+
+export const routingMiddlewareFactory: MiddlewareFactory = (coreStart, history) => {
```

Comment:
kinda disappointed that other middleware will be dependent on this one. They can't function (or be tested) alone now. And the dependency isn't explicit or tracked anywhere.